risc0-adapter/src/lib.rs

```diff
@@ -0,0 +1,28 @@
+use risc0_zkvm::sha::Digest;
+use serde::{Deserialize, Serialize};
+use sovereign_sdk::zk::traits::Matches;
+
+pub mod guest;
+#[cfg(feature = "native")]
+pub mod host;
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Risc0MethodId([u32; 8]);
+
+impl Matches<Self> for Risc0MethodId {
+    fn matches(&self, other: &Self) -> bool {
+        self.0 == other.0
```

Comment:
Why not implement automatic implementation for `PartialEq` types?

```rust
impl<T: PartialEq> Matches<T> for T {
    fn matches(&self, other: &T) -> bool {
        self == other
    }
}
```

Reply:
`Matches` is a looser concept than `PartialEq` - so a type where `a != b`  might still return `true` for `a.matches(b)`.